utils/mkdocs.py

```diff
@@ -49,16 +49,28 @@ def set_proxy():
         )
 
 
+_cached_version_branch = None
+
+
 def get_version_branch(major_version, addons_git_repo_url):
-    """Check if version branch for the current GRASS version exists,
-    if not, take branch for the previous version
-    For the official repo we assume that at least one version branch is present
+    """Get version branch for addons
+
+    Check if version branch for the current GRASS version exists,
+    if not, take branch for the previous version.
+    For the official repo, we assume that at least one version branch is present.
+
+    Getting the latest version is expensive because we need to ask online for the latest
+    branch. Hence, we cache the version branch in a global variable.
 
     :param major_version int: GRASS GIS major version
     :param addons_git_repo_url str: Addons Git repository URL
 
     :return version_branch str: version branch
     """
+    global _cached_version_branch
+    if _cached_version_branch is not None:
+        return _cached_version_branch
+
```

Comment:
Did you ever consider the `@cache` or `@lru_cache`, if they fit the use case?
https://docs.python.org/3/library/functools.html#functools.cache

https://docs.python.org/3/library/functools.html#functools.lru_cache

Choosing between both depends on how many different argument combinations you expect to be called with. `lru_cache` has a limit, and `cache` is unbounded (and is faster and lighter because of this).